src/graph/context/Iterator.cpp

```diff
@@ -8,12 +8,34 @@
 
 #include "common/datatypes/Edge.h"
 #include "common/datatypes/Vertex.h"
+#include "common/memory/MemoryUtils.h"
 #include "graph/util/SchemaUtil.h"
 #include "interface/gen-cpp2/common_types.h"
+
+DECLARE_int32(num_rows_to_check_memory);
+DECLARE_double(system_memory_high_watermark_ratio);
+
 namespace nebula {
 namespace graph {
-GetNeighborsIter::GetNeighborsIter(std::shared_ptr<Value> value)
-    : Iterator(value, Kind::kGetNeighbors) {
+
+bool Iterator::hitsSysMemoryHighWatermark() const {
+  if (checkMemory_) {
+    if (numRowsModN_ >= FLAGS_num_rows_to_check_memory) {
+      numRowsModN_ -= FLAGS_num_rows_to_check_memory;
+    }
+    if (UNLIKELY(numRowsModN_ == 0)) {
+      if (MemoryUtils::kHitMemoryHighWatermark.load()) {
+        throw std::runtime_error(
+            folly::sformat("Used memory hits the high watermark({}) of total system memory.",
+                           FLAGS_system_memory_high_watermark_ratio));
```

Comment:
Where is handler the runtime error?

Reply:
> Where is handler the runtime error?

What's about as below ?
```
if (MemoryUtils::kHitMemoryHighWatermark.load()) {
LOG(ERROR);
return true;
}
```

Reply:
There is a thenError in QueryInstance

Reply:
> There is a thenError in QueryInstance

got . LGTM now. Please resolved conflicts.